### DISCUSSION TITLE
Why sam throws Error: AWS Credentials are required. Please configure your credentials. for root user

### DISCUSSION THREAD

**Discussion by sarvpriy** [aws/aws-sam-cli#4712] [Questions] [answered]
```bash
$ sudo sam validate
2023-02-14 15:31:49 Loading policies from IAM...
Error: AWS Credentials are required. Please configure your credentials.
$ sam validate
2023-02-14 15:31:54 Loading policies from IAM...
2023-02-14 15:32:04 Finished loading policies from IAM.
/home/new-sam-app/template.yaml is a valid SAM Template
$ sudo sam validate
2023-02-14 15:32:13 Loading policies from IAM...
Error: AWS Credentials are required. Please configure your credentials
```

**Comment by jfuss** [comment, marked as answer]
@sarvpriy You need to provide credentials. This can be done through setting the [Environment Variables](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#environment-variables) (we use boto3 under the hood) or providing `--profile` to the command.

**Reply by sarvpriy** [reply]
aws credentials are already added. I am asking about the strange behaviour of using `sam cli` as a root user

**Reply by jfuss** [reply]
How are aws credentials added?

SAM CLI doesn't need root permissions, so the easy thing here is just not using `sudo` at all.
